src/python/pants/backend/python/providers/python_build_standalone/rules.py

```diff
@@ -69,6 +75,232 @@ class PBSPythonInfo(TypedDict):
 PBSVersionsT = dict[str, dict[str, dict[str, PBSPythonInfo]]]
 
 
+@dataclass
+class _ParsedPBSPython:
+    py_version: Version
+    pbs_release_tag: Version
+    platform: Platform
+    url: str
+    sha256: str
+    size: int
+
+
+def _parse_py_version_and_pbs_release_tag(
+    version_and_tag: str,
+) -> tuple[Version | None, Version | None]:
+    version_and_tag = version_and_tag.strip()
+    if not version_and_tag:
+        return None, None
+
+    parts = version_and_tag.split("+", 1)
+    py_version: Version | None = None
+    pbs_release_tag: Version | None = None
+
+    if len(parts) >= 1:
+        try:
+            py_version = Version(parts[0])
+        except InvalidVersion:
+            raise ValueError(f"Version `{parts[0]}` is not a valid Python version.")
+
+    if len(parts) == 2:
+        try:
+            pbs_release_tag = Version(parts[1])
+        except InvalidVersion:
+            raise ValueError(f"PBS release tag `{parts[1]}` is not a valid version.")
+
+    return py_version, pbs_release_tag
+
+
+def _parse_pbs_url(url: str) -> tuple[Version, Version, Platform]:
+    parsed_url = urllib.parse.urlparse(urllib.parse.unquote(url))
+    base_path = posixpath.basename(parsed_url.path)
+
+    base_path_no_prefix = base_path.removeprefix("cpython-")
+    if base_path_no_prefix == base_path:
+        raise ValueError(
+            f"Unable to parse the provided URL since it does not have a cpython prefix as per the PBS naming convention: {url}"
+        )
+
+    base_path_parts = base_path_no_prefix.split("-", 1)
+    if len(base_path_parts) != 2:
+        raise ValueError(
+            f"Unable to parse the provided URL because it does not follow the PBS naming convention: {url}"
+        )
+
+    py_version, pbs_release_tag = _parse_py_version_and_pbs_release_tag(base_path_parts[0])
+    if not py_version or not pbs_release_tag:
+        raise ValueError(
+            "Unable to parse the Python version and PBS release tag from the provided URL "
+            f"because it does not follow the PBS naming convention: {url}"
+        )
+
+    platform: Platform
+    match base_path_parts[1].split("-"):
+        case ["x86_64", "unknown", "linux", "gnu", *_]:
```

Comment:
Should/could we accept the `musl` URLs here too? E.g. https://github.com/astral-sh/python-build-standalone/releases/download/20241219/cpython-3.10.16+20241219-x86_64-unknown-linux-musl-install_only.tar.gz

From the perspective of Pants, I think either one works fine.

Reply:
Good spot. Will add it.

Reply:
@huonw: By the same thought, we should probably also infer the `x86_64_v2`, `x86_64_v3`, and`x86_64_v4` architecture variants?

Reply:
Oh, I hadn't noticed those. Yeah, it'd make sense to me to categorise them as x86-64 too, if the user has supplied those URLs.